Transformers: Support inner vs outer join (#53913)
grafana/grafana: docs/sources/panels/transform-data/index.md

> This transformation allows you to extract some key information out of your time series and display them in a convenient way.

Reviewer: ```suggestion
This transformation enables you to extract key information from your time series and display it in a convenient way.
```

### Join by field

Use this transformation to join multiple results into a single table. This is especially useful for converting multiple
time series results into a single wide table with a shared time field.

#### Inner join

> An inner join will merge data from multiple tables where all tables share the same value from the selected field. This will exclude

Reviewer: ```suggestion
An inner join merges data from multiple tables where all tables share the same value from the selected field. This type of join excludes
```

data where values do not match in every result.

> Use this transformation to combine the results from multiple queries (combining on a passed join field or the first time column) into one single result and drop rows where a successful join isn't able to occur - performing an inner join.

Reviewer: ```suggestion
Use this transformation to combine the results from multiple queries (combining on a passed join field or the first time column) into one result, and drop rows where a successful join cannot occur - performing an inner join.
```

> In the example below, we have two queries returning table data. It is visualized as two separate tables before applying the inner join transformation.

Reviewer: ```suggestion
In the following example, two queries return table data. It is visualized as two separate tables before applying the inner join transformation.
```

Query A:

| Time                | Job     | Uptime    |
| ------------------- | ------- | --------- |
| 2020-07-07 11:34:20 | node    | 25260122  |
| 2020-07-07 11:24:20 | postgre | 123001233 |
| 2020-07-07 11:14:20 | postgre | 345001233 |

Query B:

| Time                | Server   | Errors |
| ------------------- | -------- | ------ |
| 2020-07-07 11:34:20 | server 1 | 15     |
| 2020-07-07 11:24:20 | server 2 | 5      |
| 2020-07-07 11:04:20 | server 3 | 10     |

> Result after applying the inner join transformation:

Reviewer: ```suggestion
The result after applying the inner join transformation looks like the following:
```

| Time                | Job     | Uptime    | Server   | Errors |
| ------------------- | ------- | --------- | -------- | ------ |
| 2020-07-07 11:34:20 | node    | 25260122  | server 1 | 15     |
| 2020-07-07 11:24:20 | postgre | 123001233 | server 2 | 5      |

#### Outer join

> An outer join will include all data from an inner join, but will also include rows where values do not match in every input.

Reviewer: ```suggestion
An outer join includes all data from an inner join and rows where values do not match in every input.
```

> Use this transformation to combine the results from multiple queries (combining on a passed join field or the first time column) into one single result and drop rows where a successful join isn't able to occur - performing an inner join.

Reviewer: ```suggestion
Use this transformation to combine the results from multiple queries (combining on a passed join field or the first time column) into one result, and drop rows where a successful join cannot occur - performing an inner join.
```

> Use this transformation to combine the results from multiple queries (combining on a passed join field or the first time column) into one single result and drop rows where a successful join isn't able to occur - performing an inner join.

Reviewer: I'm not sure what the last bit means " - performing an inner join." Should this read, ...where a successful join cannot occur, which perform an inner join"?

Author: I fixed it :)

> In the example below, we have two queries returning table data. It is visualized as two separate tables before applying the inner join transformation.

Reviewer: ```suggestion
In the following example, two queries return table data. It is visualized as two tables before applying the inner join transformation.
```

Reviewer 2: Isn't this section about applying an `outer join` transformation vs `inner`?

Query A:

| Time                | Job     | Uptime    |
| ------------------- | ------- | --------- |
| 2020-07-07 11:34:20 | node    | 25260122  |
| 2020-07-07 11:24:20 | postgre | 123001233 |
| 2020-07-07 11:14:20 | postgre | 345001233 |

Query B:

| Time                | Server   | Errors |
| ------------------- | -------- | ------ |
| 2020-07-07 11:34:20 | server 1 | 15     |
| 2020-07-07 11:24:20 | server 2 | 5      |
| 2020-07-07 11:04:20 | server 3 | 10     |

> Result after applying the inner join transformation:

Reviewer: ```suggestion
The result after applying the inner join transformation looks like the following:
```

| Time                | Job     | Uptime    | Server   | Errors |
| ------------------- | ------- | --------- | -------- | ------ |
| 2020-07-07 11:34:20 | node    | 25260122  | server 1 | 15     |
| 2020-07-07 11:24:20 | postgre | 123001233 | server 2 | 5      |

> In the example below, I have a template query displaying time series data from multiple servers in a table visualization. I can only view the results of one query at a time.

Reviewer: ```suggestion
In the following example, a template query displays time series data from multiple servers in a table visualization. The results of only one query can be viewed at a time.
```
